api/src/test/java/uk/gov/dhsc/htbhf/claimant/utilities/DatabasePopulator.java

```diff
@@ -97,19 +124,53 @@ void generateClaimsAndPaymentCycles() {
             EligibilityStatus paymentEligibility = PAYMENT_ELIGIBILITY_STATUS.get(status);
             String ninoPrefix = NINO_PREFIX.get(status);
             Query query = entityManager.createNativeQuery("select count(*) from claimant where left(nino , 2) = '" + ninoPrefix + "'");
-            int ninoStartIndex = ((Number)query.getSingleResult()).intValue() + 1;
+            int ninoStartIndex = ((Number) query.getSingleResult()).intValue() + 10001; // ensure we have a non-zero number in the first 2 digits
             log.error("Creating {} claims in status {}, starting from {}", claims, status, ninoStartIndex);
+            List<Callable<Long>> jobs = new ArrayList<>(claims);
             for (int i = 0; i < claims; i++) {
-                Claim claim = createClaim(status, eligibilityStatus, ninoPrefix, ninoStartIndex + i);
-                if (paymentEligibility != null) {
-                    createPaymentCycle(cycleDuration, paymentEligibility, claim);
-                }
-                if (countCreated.incrementAndGet() % 100 == 0) {
-                    log.error("Created {} claims so far", countCreated.get());
-                }
-
+                Integer ninoNumber = ninoStartIndex + i;
+                Callable<Long> job = () -> {
```

Comment:
I know this is only test code, but could we extra this code to declutter this large method an make it easier to understand?

Reply:
I did look at doing so, but there are seven parameters we'd need to pass.

Reply:
Extracted a NewClaimDetails object and method to create the job.